fractal/parser.py

```diff
@@ -511,11 +528,19 @@
     type=int,
     help="Workflow ID",
 )
-workflow_add_task_parser.add_argument(
-    "task_id_or_name", help="ID or name of the new task", type=str
+
+workflow_add_task_id_or_name_group = (
+    workflow_add_task_parser.add_mutually_exclusive_group(required=True)
 )
+workflow_add_task_id_or_name_group.add_argument(
+    "--task-id", help="ID of the task to add", type=int
+)
+workflow_add_task_id_or_name_group.add_argument(
+    "--task-name", help="Name of the task to add", type=str
+)
+
 workflow_add_task_parser.add_argument(
-    "--version", type=str, help="Version of the new task"
+    "--task-version", type=str, help="Version of task to add"
```

Comment:
Add to version help message something like "(only accepted in combination with --task-name)"